src/Components/test/E2ETest/ServerExecutionTests/ComponentWithParametersTest.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using TestServer;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests
+{
+    public class ComponentWithParametersTest : ServerTestBase<BasicTestAppServerSiteFixture<PrerenderedStartup>>
+    {
+        public ComponentWithParametersTest(
+            BrowserFixture browserFixture,
+            BasicTestAppServerSiteFixture<PrerenderedStartup> serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture, output)
+        {
+        }
+
+        [Fact]
+        public void PassingParametersToComponentsWorks()
+        {
+            Navigate("/prerendered/componentwithparameters?QueryValue=testQueryValue");
+
+            BeginInteractivity();
+
+            Browser.Exists(By.CssSelector(".interactive"));
+
+            var parameter1 = Browser.FindElement(By.CssSelector(".Param1"));
+            Assert.Equal(100, parameter1.FindElements(By.CssSelector("li")).Count);
+            Assert.Equal("99 99", parameter1.FindElement(By.CssSelector("li:last-child")).Text);
+
+            // The assigned value is of a more derived type than the declared model type. This check
+            // verifies we use the actual model type during round tripping.
+            var parameter2 = Browser.FindElement(By.CssSelector(".Param2"));
+            Assert.Equal("Value Derived-Value", parameter2.Text);
+
+            // This check verifies CaptureUnmatchedValues works
```

Comment:
Why are we validating this here, it feels completely unrelated to the feature or to the ability to pass parameters to the root component?

Reply:
Steve noted in the other PR - https://github.com/aspnet/AspNetCore/pull/14465#discussion_r329495458, that we want to ensure we have the ability to pass CaptureUnmatchedValues around. I wanted to make sure we covered that in the context of using the tag helper.